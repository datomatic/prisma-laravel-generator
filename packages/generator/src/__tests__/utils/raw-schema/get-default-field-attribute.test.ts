import {getSample} from '../../__fixtures__/get-sample';
import getDefaultFieldAttribute from '../../../utils/raw-schema/get-default-field-attribute';

test('getDefaultFieldAnnotation', async () => {
  const {dmmf, raw} = await getSample();

  for (const modelInfo of dmmf.datamodel.models) {
    for (const fieldInfo of modelInfo.fields) {
      expect(
        getDefaultFieldAttribute(fieldInfo.name, modelInfo.name, raw),
      ).toMatchSnapshot(`${modelInfo.name} - ${fieldInfo.name}`);
    }
  }
});
