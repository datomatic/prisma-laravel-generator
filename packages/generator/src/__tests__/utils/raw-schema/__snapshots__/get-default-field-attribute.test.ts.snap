// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDefaultFieldAnnotation: Category - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Category - name 1`] = `false`;

exports[`getDefaultFieldAnnotation: Category - posts 1`] = `false`;

exports[`getDefaultFieldAnnotation: DefaultValues - age 1`] = `"18"`;

exports[`getDefaultFieldAnnotation: DefaultValues - bool 1`] = `"true"`;

exports[`getDefaultFieldAnnotation: DefaultValues - email 1`] = `"\\"hello@world.test\\""`;

exports[`getDefaultFieldAnnotation: DefaultValues - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: DefaultValues - name 1`] = `"\\"N.A.\\""`;

exports[`getDefaultFieldAnnotation: DefaultValues - now 1`] = `"now()"`;

exports[`getDefaultFieldAnnotation: DefaultValues - role 1`] = `"USER"`;

exports[`getDefaultFieldAnnotation: DefaultValues - uuid 1`] = `"uuid()"`;

exports[`getDefaultFieldAnnotation: DefaultValuesWithoutMethodCall - age 1`] = `"18"`;

exports[`getDefaultFieldAnnotation: DefaultValuesWithoutMethodCall - bool 1`] = `"true"`;

exports[`getDefaultFieldAnnotation: DefaultValuesWithoutMethodCall - email 1`] = `"\\"hello@world.test\\""`;

exports[`getDefaultFieldAnnotation: DefaultValuesWithoutMethodCall - id 1`] = `false`;

exports[`getDefaultFieldAnnotation: DefaultValuesWithoutMethodCall - name 1`] = `"\\"N.A.\\""`;

exports[`getDefaultFieldAnnotation: EagerLoadA - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: EagerLoadA - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: EagerLoadB - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: EagerLoadB - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: EagerLoadB - relatedId 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA2 - explicit_many_to_many_b2 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA2 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA3 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA3 - otherIdA 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA3 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA4 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA4 - otherIdA 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA4 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA5 - hasOne 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA5 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA5 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA6 - key 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA6 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA7 - key 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyA7 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB2 - explicit_many_to_many_a2 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB2 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB3 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB3 - otherIdB 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB3 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB4 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB4 - otherIdB 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB4 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB5 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB5 - pivot 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB5 - pivot_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB5 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB6 - key 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB6 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB7 - key 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyB7 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot - A 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot - A_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot - B 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot - B_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot2 - explicit_many_to_many_a2 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot2 - explicit_many_to_many_a2_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot2 - explicit_many_to_many_b2 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot2 - explicit_many_to_many_b2_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot2 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot3 - A 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot3 - A_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot3 - B 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot3 - B_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot3 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot4 - A 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot4 - A_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot4 - B 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot4 - B_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot4 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot5 - A 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot5 - A_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot5 - A2 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot5 - A2_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot5 - B 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot5 - B_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot5 - B2 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot5 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot6 - A 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot6 - A_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot6 - B 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot6 - B_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot6 - key 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot7 - A 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot7 - A_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot7 - B 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot7 - B_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: ExplicitManyToManyPivot7 - key 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: FieldComments - age 1`] = `false`;

exports[`getDefaultFieldAnnotation: FieldComments - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: FieldComments - height 1`] = `false`;

exports[`getDefaultFieldAnnotation: FieldComments - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: FieldComments - name 1`] = `false`;

exports[`getDefaultFieldAnnotation: FieldMultipleComments - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: FieldMultipleComments - name 1`] = `false`;

exports[`getDefaultFieldAnnotation: Fillable - age 1`] = `"18"`;

exports[`getDefaultFieldAnnotation: Fillable - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: Fillable - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Fillable - name 1`] = `false`;

exports[`getDefaultFieldAnnotation: Fillable - password 1`] = `false`;

exports[`getDefaultFieldAnnotation: Guarded - age 1`] = `"18"`;

exports[`getDefaultFieldAnnotation: Guarded - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: Guarded - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Guarded - name 1`] = `false`;

exports[`getDefaultFieldAnnotation: Guarded - password 1`] = `false`;

exports[`getDefaultFieldAnnotation: Hidden - api_key 1`] = `false`;

exports[`getDefaultFieldAnnotation: Hidden - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: Hidden - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Hidden - name 1`] = `false`;

exports[`getDefaultFieldAnnotation: Hidden - password 1`] = `false`;

exports[`getDefaultFieldAnnotation: Ignore - age 1`] = `false`;

exports[`getDefaultFieldAnnotation: Ignore - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: Ignore - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ImplicitManyToManyA - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ImplicitManyToManyA - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: ImplicitManyToManyB - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ImplicitManyToManyB - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: MappedId - firstname 1`] = `false`;

exports[`getDefaultFieldAnnotation: MappedId - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: MassAssignable - age 1`] = `"18"`;

exports[`getDefaultFieldAnnotation: MassAssignable - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: MassAssignable - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: MassAssignable - name 1`] = `false`;

exports[`getDefaultFieldAnnotation: MassAssignable - password 1`] = `false`;

exports[`getDefaultFieldAnnotation: ModelComments - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ModelCustomPath - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ModelMultipleComments - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: MultipleComments - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: MultipleComments - password 1`] = `false`;

exports[`getDefaultFieldAnnotation: MultipleUnique - age 1`] = `false`;

exports[`getDefaultFieldAnnotation: MultipleUnique - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: MultipleUnique - firstname 1`] = `false`;

exports[`getDefaultFieldAnnotation: MultipleUnique - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: MultipleUnique - lastname 1`] = `false`;

exports[`getDefaultFieldAnnotation: MultipleUnique2 - age 1`] = `false`;

exports[`getDefaultFieldAnnotation: MultipleUnique2 - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: MultipleUnique2 - firstname 1`] = `false`;

exports[`getDefaultFieldAnnotation: MultipleUnique2 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: MultipleUnique2 - lastname 1`] = `false`;

exports[`getDefaultFieldAnnotation: NoPrimaryKey - age 1`] = `false`;

exports[`getDefaultFieldAnnotation: NoPrimaryKey - name 1`] = `false`;

exports[`getDefaultFieldAnnotation: NullableField - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: NullableField - name 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToManyA - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToManyA - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToManyA2 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToManyA2 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToManyA3 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToManyA3 - otherId 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToManyA3 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToManyA4 - main_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToManyA4 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToManyB - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToManyB - one_to_one_a2 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToManyB - relatedId 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToManyB2 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToManyB2 - one_to_many_a2 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToManyB2 - one_to_many_a2_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToManyB3 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToManyB3 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToManyB3 - relatedId 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToManyB4 - main_id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToManyB4 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToManyB4 - relatedId 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneA - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToOneA - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneA2 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToOneA2 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneA3 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToOneA3 - otherId 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneA3 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneA4 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToOneA4 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneA5 - main_id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToOneA5 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneB - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToOneB - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneB - relatedId 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneB2 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToOneB2 - one_to_one_a2 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneB2 - one_to_one_a2_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneB3 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToOneB3 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneB3 - related_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneB4 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToOneB4 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneB4 - relatedId 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneB5 - main_id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: OneToOneB5 - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: OneToOneB5 - relatedId 1`] = `false`;

exports[`getDefaultFieldAnnotation: PivotWithCustomExtend - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Post - author 1`] = `false`;

exports[`getDefaultFieldAnnotation: Post - authorId 1`] = `false`;

exports[`getDefaultFieldAnnotation: Post - categories 1`] = `false`;

exports[`getDefaultFieldAnnotation: Post - createdAt 1`] = `false`;

exports[`getDefaultFieldAnnotation: Post - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Post - published 1`] = `"false"`;

exports[`getDefaultFieldAnnotation: Post - title 1`] = `false`;

exports[`getDefaultFieldAnnotation: Post - type 1`] = `false`;

exports[`getDefaultFieldAnnotation: Post - updatedAt 1`] = `false`;

exports[`getDefaultFieldAnnotation: Profile - bio 1`] = `false`;

exports[`getDefaultFieldAnnotation: Profile - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Profile - user 1`] = `false`;

exports[`getDefaultFieldAnnotation: Profile - userId 1`] = `false`;

exports[`getDefaultFieldAnnotation: ReadOnlyField - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: ReadOnlyField - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ReadOnlyField - name 1`] = `false`;

exports[`getDefaultFieldAnnotation: ReadOnlyField - password 1`] = `false`;

exports[`getDefaultFieldAnnotation: ReadOnlyField - phone 1`] = `false`;

exports[`getDefaultFieldAnnotation: ReadonlyDatetime - editable 1`] = `false`;

exports[`getDefaultFieldAnnotation: ReadonlyDatetime - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: ReadonlyDatetime - readonly 1`] = `false`;

exports[`getDefaultFieldAnnotation: SelfRelation - child 1`] = `false`;

exports[`getDefaultFieldAnnotation: SelfRelation - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: SelfRelation - parent 1`] = `false`;

exports[`getDefaultFieldAnnotation: SelfRelation - relatedId 1`] = `false`;

exports[`getDefaultFieldAnnotation: SoftDelete1 - deleted_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: SoftDelete1 - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: SoftDelete1 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: SoftDelete1 - password 1`] = `false`;

exports[`getDefaultFieldAnnotation: SoftDelete2 - deleted_timestamp 1`] = `false`;

exports[`getDefaultFieldAnnotation: SoftDelete2 - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: SoftDelete2 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: SoftDelete2 - password 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps1 - created_timestamp 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps1 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Timestamps1 - updated_timestamp 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps2 - created_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps2 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Timestamps2 - updated_timestamp 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps3 - created_timestamp 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps3 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Timestamps3 - updated_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps4 - created_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps4 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Timestamps4 - updated_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps5 - created_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps5 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Timestamps5 - updated_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps6 - created_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps6 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Timestamps6 - updated_timestamp 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps7 - created_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps7 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Timestamps7 - updated_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps8 - created_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps8 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Timestamps8 - updated_timestamp 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps9 - created_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps9 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Timestamps9 - updated_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps10 - created_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps10 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Timestamps10 - updated_timestamp 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps11 - created_timestamp 1`] = `false`;

exports[`getDefaultFieldAnnotation: Timestamps11 - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Timestamps11 - updated_at 1`] = `false`;

exports[`getDefaultFieldAnnotation: TouchingA - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: TouchingA - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: TouchingB - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: TouchingB - related 1`] = `false`;

exports[`getDefaultFieldAnnotation: TouchingB - relatedId 1`] = `false`;

exports[`getDefaultFieldAnnotation: Types - active 1`] = `false`;

exports[`getDefaultFieldAnnotation: Types - age 1`] = `false`;

exports[`getDefaultFieldAnnotation: Types - birth 1`] = `false`;

exports[`getDefaultFieldAnnotation: Types - blob 1`] = `false`;

exports[`getDefaultFieldAnnotation: Types - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Types - name 1`] = `false`;

exports[`getDefaultFieldAnnotation: Types - price 1`] = `false`;

exports[`getDefaultFieldAnnotation: Types - schema 1`] = `false`;

exports[`getDefaultFieldAnnotation: Types - seconds 1`] = `false`;

exports[`getDefaultFieldAnnotation: Types - speed 1`] = `false`;

exports[`getDefaultFieldAnnotation: Unique - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: Unique - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Unique - name 1`] = `false`;

exports[`getDefaultFieldAnnotation: Unique - username 1`] = `false`;

exports[`getDefaultFieldAnnotation: UnsupportedTypes - id 1`] = `false`;

exports[`getDefaultFieldAnnotation: User - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: User - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: User - name 1`] = `false`;

exports[`getDefaultFieldAnnotation: User - posts 1`] = `false`;

exports[`getDefaultFieldAnnotation: User - profile 1`] = `false`;

exports[`getDefaultFieldAnnotation: User - role 1`] = `"USER"`;

exports[`getDefaultFieldAnnotation: Visible - api_key 1`] = `false`;

exports[`getDefaultFieldAnnotation: Visible - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: Visible - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: Visible - name 1`] = `false`;

exports[`getDefaultFieldAnnotation: Visible - password 1`] = `false`;

exports[`getDefaultFieldAnnotation: WithArray - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: WithArray - numbers 1`] = `false`;

exports[`getDefaultFieldAnnotation: WithBackedEnum - array 1`] = `false`;

exports[`getDefaultFieldAnnotation: WithBackedEnum - defaultEnum 1`] = `"A"`;

exports[`getDefaultFieldAnnotation: WithBackedEnum - enum 1`] = `false`;

exports[`getDefaultFieldAnnotation: WithBackedEnum - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: WithBackedEnum - nullable 1`] = `false`;

exports[`getDefaultFieldAnnotation: WithCustomExtend - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: WithDbType - email 1`] = `false`;

exports[`getDefaultFieldAnnotation: WithDbType - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: WithEnum - array 1`] = `false`;

exports[`getDefaultFieldAnnotation: WithEnum - defaultEnum 1`] = `"A"`;

exports[`getDefaultFieldAnnotation: WithEnum - enum 1`] = `false`;

exports[`getDefaultFieldAnnotation: WithEnum - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: WithEnum - nullable 1`] = `false`;

exports[`getDefaultFieldAnnotation: WithGlobalPrimaryKey - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: WithGlobalPrimaryKeyDifferentName - primary_id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: WithGlobalStringPrimaryKey - id 1`] = `false`;

exports[`getDefaultFieldAnnotation: WithGlobalStringPrimaryKeyDifferentName - primary_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: WithInterface - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: WithMultipleInterfaces - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: WithMultipleTraits - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: WithPrimaryKey - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: WithPrimaryKeyDifferentName - primary_id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: WithStringPrimaryKey - id 1`] = `false`;

exports[`getDefaultFieldAnnotation: WithStringPrimaryKeyDifferentName - primary_id 1`] = `false`;

exports[`getDefaultFieldAnnotation: WithTableMap - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: WithTrait - id 1`] = `"autoincrement()"`;

exports[`getDefaultFieldAnnotation: snake_cased_models - firstname 1`] = `false`;

exports[`getDefaultFieldAnnotation: snake_cased_models - id 1`] = `"autoincrement()"`;
